src/main/java/com/iota/iri/zmq/TimeWindowedApproveeCounter.java

```diff
@@ -0,0 +1,93 @@
+package com.iota.iri.zmq;
+
+import com.iota.iri.controllers.TransactionViewModel;
+import com.iota.iri.model.Hash;
+import com.iota.iri.storage.Tangle;
+import com.iota.iri.utils.dag.DAGHelper;
+import com.iota.iri.utils.dag.TraversalException;
+
+import java.util.HashSet;
+
+/**
+ * Utility class to count the number of distinct approvees that have an arrival time within the given time window.
+ */
+class TimeWindowedApproveeCounter {
+
+    private final Tangle tangle;
+    private final long minTransactionAgeSeconds;
+    private final long maxTransactionAgeSeconds;
+
+    TimeWindowedApproveeCounter(Tangle tangle, long minTransactionAgeSeconds, long maxTransactionAgeSeconds) {
+        this.tangle = tangle;
+        this.minTransactionAgeSeconds = minTransactionAgeSeconds;
+        this.maxTransactionAgeSeconds = maxTransactionAgeSeconds;
+    }
+
+    private boolean continueTraversal(long now, TransactionViewModel transactionViewModel) {
```

Comment:
suggest renaming to `isOlderThanMaxAge`, more descriptive

Reply:
Also rename `transactionViewModel` to `transaction` for consistency with the function below